pkg/services/ngalert/store/image.go

```diff
@@ -0,0 +1,95 @@
+package store
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/gofrs/uuid"
+
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+)
+
+var (
+	// ErrImageNotFound is returned when the image does not exist.
+	ErrImageNotFound = errors.New("image not found")
+)
+
+type Image struct {
+	ID        int64     `xorm:"pk autoincr 'id'"`
+	Token     string    `xorm:"token"`
+	Path      string    `xorm:"path"`
+	URL       string    `xorm:"url"`
+	CreatedAt time.Time `xorm:"created_at"`
+	ExpiresAt time.Time `xorm:"expires_at"`
+}
+
+// A XORM interface that lets us clean up our SQL session definition.
+func (i *Image) TableName() string {
+	return "alert_image"
+}
+
+type ImageStore interface {
+	// Get returns the image with the token or ErrImageNotFound.
+	GetImage(ctx context.Context, token string) (*Image, error)
+
+	// Saves the image or returns an error.
+	SaveImage(ctx context.Context, img *Image) error
+}
+
+func (st DBstore) GetImage(ctx context.Context, token string) (*Image, error) {
+	var img Image
+	if err := st.SQLStore.WithDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		exists, err := sess.Where("token = ?", token).Get(&img)
+		if err != nil {
+			return fmt.Errorf("failed to get image: %w", err)
+		}
+		if !exists {
+			return ErrImageNotFound
+		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+	return &img, nil
+}
+
+func (st DBstore) SaveImage(ctx context.Context, img *Image) error {
+	return st.SQLStore.WithTransactionalDbSession(ctx, func(sess *sqlstore.DBSession) error {
+		// TODO: Is this a good idea?
+		img.ExpiresAt = TimeNow().Add(1 * time.Minute).UTC()
```

Comment:
Yes it is. I think if we're OK with the value I should just make it a constant.

Reply:
This actually isn't that. It's the expiration time that's set on a database row. `DeleteExpiredTokens` isn't actually called here yet, which is the only code that uses it. I didn't want to set it up, since a token row is tiny and the worst thing that's going to happen is a "Not Found" if you look up an old one.

 I'll expand the comment and link an issue for followup.